mcs/mcs/statement.cs

```diff
@@ -2677,7 +2677,7 @@ public static string GetCompilerGeneratedName (Block block)
 
 		public string GetReadOnlyContext ()
 		{
-			switch (flags & Flags.ReadonlyMask) {
+			switch (flags & (Flags.ForeachVariable | Flags.FixedVariable | Flags.UsingVariable)) {
```

Comment:
this change isn't necessary, and actually shouldn't be done, otherwise you have to revisit this code if you add another `ReadonlyMask` flag

Reply:
That's the intention, the switch has to be updated when another ReadonlyMask is used